commercetools-models/src/main/java/io/sphere/sdk/customobjects/commands/CustomObjectDeleteCommandImpl.java

```diff
@@ -30,29 +30,49 @@ public HttpRequestIntent httpRequestIntent() {
         return HttpRequestIntent.of(HttpMethod.DELETE, CustomObjectEndpoint.PATH + path);
     }
 
-    CustomObjectDeleteCommandImpl(final String container, final String key, final Class<T> valueClass) {
-        this(container, key, SphereJsonUtils.convertToJavaType(valueClass));
+    CustomObjectDeleteCommandImpl(final String container, final String key,final boolean eraseData, final JavaType valueJavaType) {
+        this.path = format("/%s/%s?%s", container, key,eraseData? "dataErasure=true":"");
+        this.javaType = CustomObjectUtils.getCustomObjectJavaTypeForValue(valueJavaType);
     }
 
-    CustomObjectDeleteCommandImpl(final String container, final String key, final TypeReference<T> typeReference) {
-        this(container, key, SphereJsonUtils.convertToJavaType(typeReference));
+    CustomObjectDeleteCommandImpl(final String id, final Long version,final boolean eraseData , final JavaType valueJavaType) {
+        this.path = format("/%s?version=%d%s", id, version,eraseData? "&dataErasure=true":"");
+        this.javaType = CustomObjectUtils.getCustomObjectJavaTypeForValue(valueJavaType);
     }
 
-    CustomObjectDeleteCommandImpl(final String container, final String key, final JavaType valueJavaType) {
-        this.path = format("/%s/%s", container, key);
-        this.javaType = CustomObjectUtils.getCustomObjectJavaTypeForValue(valueJavaType);
+    CustomObjectDeleteCommandImpl(final String container, final String key, final Class<T> valueClass) {
+        this(container, key,true, SphereJsonUtils.convertToJavaType(valueClass));
```

Comment:
According to our dev documentation, the default for `eraseData` is `false`. And I would expect that our JVM SDK uses this default here too.